tests/lib/rules/no-this-in-sfc.js

```diff
@@ -217,13 +217,6 @@ ruleTester.run('no-this-in-sfc', rule, {
       return <div onClick={onClick}>{this.props.foo}</div>;
     }`,
     errors: [{message: ERROR_MESSAGE}, {message: ERROR_MESSAGE}]
-  }, {
-    code: `
-    () => {
-      this.something();
-      return null;
-    }`,
-    errors: [{message: ERROR_MESSAGE}]
```

Comment:
This test case is moot. It is an unused expression.

Reply:
i'm not sure why; it's still a usage of `this` that someone might expect to have side effects.

Reply:
I mean the arrow function is not reference-able and the program is equivalent to no-op, therefore it should never happen in real world code.

Reply:
ah fair, since the arrow function won't be recognized as a component.